etna/models/seasonal_ma.py

```diff
@@ -128,15 +146,18 @@ def forecast(self, ts: TSDataset, prediction_size: int, return_components: bool
         ValueError:
             if forecast context contains NaNs
         """
-        if return_components:
-            raise NotImplementedError("This mode isn't currently implemented!")
-
         df = ts.to_pandas()
-        new_df = self._forecast(df=df, prediction_size=prediction_size)
-        ts.df = new_df
+        y_pred = self._forecast(df=df, prediction_size=prediction_size)
+        ts.df = ts.df.iloc[-prediction_size:]
+        ts.df.loc[:, pd.IndexSlice[:, "target"]] = y_pred
+
+        if return_components:
+            df.loc[-prediction_size:, pd.IndexSlice[:, "target"]] = y_pred
```

Comment:
We still use `df.loc[-prediction_size:, pd.IndexSlice[:, "target"]]`, this probably won't work on all pandas versions (we have error in testing under different pandas versions).

The `:prediction_size` implies that first index is integer, but it isn't.

Reply:
Sorry, missed this place, you are right. Fixed it

Reply:
May be we should make it more clear why in `forecast` we add `y_pred` into `df`. Because autoregression logic lies within `def _forecast` and isn't present on level of `def forecast`.

Reply:
Or we should make it more clear what kind of `df` we are going to use inside `_predict_components`. We require that it contains lags that was used to make a prediction (taking into account auto-regression), it isn't clear.

Reply:
Add docstring to `_predict_components` with description of `df`